mypy.ini

```diff
@@ -0,0 +1,8 @@
+# SPDX-FileCopyrightText: 2022 Matt Land
+#
+# SPDX-License-Identifier: Unlicense
+[mypy]
+python_version = 3.7
+disallow_untyped_defs = True
+disable_error_code = no-redef
+exclude = (examples|tests|setup.py|docs)
```

Comment:
I don't think we are ready to add the mypy configuration file yet. Checking in with @kattni too though, do you have preference on adding this file now vs. waiting until whenever we are ready to enable mypy in our CI?

Reply:
I can pull it, NBD. But as a (sometimes) maintainer, I'd be pleased it if it went in, because if/ when the next person works here (with types), they would immediately need to recreate this file (or another file providing similar functionality).

I don't think this mypy.ini is ready to go to Cookie Cutter for all repos yet:
- run 5-6 projects and see how similar their mypy.ini files. Bake a  common target config file from learnings.
- standardize the target config in mypy.ini, pre-commit hooks, or setup.cfg